test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -0,0 +1,423 @@
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent.get(), 960000 /* milliseconds */))
+    {
+        channelOperation.Cancel();
+        channelOperationResult = ERROR_TIMEOUT;
+    }
+    else
+    {
+        channelOperation.Close(); // Do not call getresults after this
+    }
+
+    return (channelOperationResult == S_OK);
+}
+
+bool MultipleChannelRequestUsingSameRemoteId()
+{
+    wil::unique_handle channelEvent1 = wil::unique_handle(CreateEvent(nullptr, FALSE, FALSE, nullptr));
+    wil::unique_handle channelEvent2 = wil::unique_handle(CreateEvent(nullptr, FALSE, FALSE, nullptr));
+
+    auto channelOperationResult2 = S_OK;
+
+    auto channelOperation1 = PushNotificationManager::CreateChannelAsync(remoteId1);
+
+    channelOperation1.Completed(
+        [&channelEvent1](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+            AsyncStatus const /* asyncStatus */)
+        {
+            auto result = sender.GetResults();
+            if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+            {
+                result.Channel().Close(); // Avoids using cached channel everytime the test runs
+            }
+            else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+            {
+            }
+
+            SetEvent(channelEvent1.get());
+        });
+
+    auto channelOperation2 = PushNotificationManager::CreateChannelAsync(remoteId1);
+
+    channelOperation2.Completed(
+        [&channelEvent2, &channelOperationResult2](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+            AsyncStatus const /* asyncStatus */)
+        {
+            auto result = sender.GetResults();
+            if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+            {
+                result.Channel().Close(); // Avoids using cached channel everytime the test runs
+            }
+            else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+            {
+                channelOperationResult2 = result.ExtendedError();
+            }
+
+            SetEvent(channelEvent2.get());
+        });
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent2.get(), 960000 /* milliseconds */))
+    {
+        channelOperation2.Cancel();
+        channelOperationResult2 = ERROR_TIMEOUT;
+    }
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent1.get(), 960000 /* milliseconds */))
+    {
+        channelOperation1.Cancel();
+    }
+
+    return (channelOperationResult2 == WPN_E_OUTSTANDING_CHANNEL_REQUEST);
```

Comment:
I still don't understand why it's wrong to request the channel twice.

Reply:
The CreateChannelAsync API can only service one channel request at any point of time. If the first one is being serviced then the second one would end up seeing the WPN_E_OUTSTANDING_CHANNEL_REQUEST and this test is to verify that.